P/PROJ/build_tarballs.jl

```diff
@@ -0,0 +1,85 @@
+using BinaryBuilder
+
+name = "PROJ"
+version = v"6.2.1"
+
+# Collection of sources required to build PROJ
+sources = [
+    "https://download.osgeo.org/proj/proj-$version.tar.gz" =>
+    "7f2e0fe63312f1e766057cceb53dc9585c4a335ff6641de45696dbd40d17c340",
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/proj-*/
+
+# sqlite needed to build proj.db, so this should not be the
+# cross-compiled one since it needs to be executed on the host
+apk add sqlite
+
+if [[ ${target} == *mingw* ]]; then
+    SQLITE3_LIBRARY=${libdir}/libsqlite3-0.dll
+else
+    SQLITE3_LIBRARY=${libdir}/libsqlite3.${dlext}
+fi
+
+mkdir build
+cd build
+cmake -DCMAKE_INSTALL_PREFIX=$prefix \
+      -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+      -DSQLITE3_INCLUDE_DIR=$prefix/include \
+      -DSQLITE3_LIBRARY=$SQLITE3_LIBRARY \
+      -DHAVE_PTHREAD_MUTEX_RECURSIVE_DEFN=1 \
+      -DBUILD_LIBPROJ_SHARED=ON \
+      ..
+cmake --build .
+make install
+
+# add proj-datumgrid files directly to the result
+wget https://download.osgeo.org/proj/proj-datumgrid-1.8.tar.gz
```

Comment:
It's probably better to have this as additional source. You'll need to manually install the license file: https://juliapackaging.github.io/BinaryBuilder.jl/dev/build_tips/#Installing-the-license-file-1

Reply:
Ok! Since this is data only, should it also go in Yggdrasil or should I manually make an Artifact? Are there any examples for that?

Probably a good idea since there are more data packages for PROJ, and I may want to make some optional at some point.

Reply:
`proj-datumgrid` is optional, so no need to make it a dependency here. I removed it now. Reason we added it before is because I think one build step inadvertently assumed it was present, but now it seems to work fine without.

Then I think `PROJ_jll` can do without any `datumgrid` data, and if we want we can add it as a seperate Artifact dependency to `Proj4.jll`.